### ISSUE TITLE
Verify Dataset __getitem__ return value

### ISSUE THREAD

**Issue by Kaszanas** [Kaszanas/SC2_Datasets#4]
There is no clear documentation on what should be returned from ```__getitem__``` method using a PyTorch ```Dataset``` class.

It means a lack of clear interface for any machine learning model that is going to be used down the line.

After some research it seems that what is commonly used as a return type is ```return x, y``` where ```x``` is a tensor or numpy array containing features, and ```y``` is a tensor or numpy array containing labels.

For the case of ```SC2ReplayData``` there is no clear list of features and labels. Some kind of logic needs to be introduced on dataset initialization that allows for feature/label selection. PyTorch transforms seem to be a good way to pass such logic to the dataset.

This would imply that the rest of the logic is applied within the model.

References:

- [Understanding PyTorch with an example: a step-by-step tutorial](https://towardsdatascience.com/understanding-pytorch-with-an-example-a-step-by-step-tutorial-81fc5f8c4e8e#2e24)
- [PyTorch Dataset](https://pytorch.org/docs/stable/data.html#torch.utils.data.Dataset)
- [MNIST example](https://pytorch.org/vision/stable/_modules/torchvision/datasets/mnist.html#MNIST.__getitem__)

If you have any thoughts on that let me know:
@leafnode
